client/domain/ReinforcementBulletObjectFactory.js

```diff
@@ -0,0 +1,57 @@
+// Copyright 2017 The TIE Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Factory for creating new frontend instances of
+ * reinforcementBullet domain objects.
+ */
+
+tie.factory('ReinforcementBulletObjectFactory', [
+  function() {
+
+    var ReinforcementBullet = function(passed, content) {
+      this._passed = passed;
+      this._content = content;
+      if (passed) {
```

Comment:
this._imgName = (passed ? 'pass.png' : 'fail.png');

but then, actually, you don't even need this._imgName at all. Just add a method ReinforcementBullet.prototype.getImageFilename() and do the computation there.